src/pudl/settings.py

```diff
@@ -243,41 +244,82 @@ class Eia860Settings(GenericDatasetSettings):
         data_source: DataSource metadata object
         years: list of years to validate.
         eia860m: whether or not to incorporate an EIA-860m month.
-        eia860m_year_month ClassVar[str]: The 860m year-month to incorporate.
+        eia860m_year_months ClassVar[str]: The 860m year-months to incorporate.
     """
 
     data_source: ClassVar[DataSource] = DataSource.from_id("eia860")
     years: list[int] = data_source.working_partitions["years"]
-
     eia860m: bool = True
-    eia860m_year_month: ClassVar[str] = max(
-        DataSource.from_id("eia860m").working_partitions["year_months"]
-    )
+    all_eia860m_year_months: list[str] = DataSource.from_id(
+        "eia860m"
+    ).working_partitions["year_months"]
+    eia860m_year_months: list[str] = [max(all_eia860m_year_months)]
 
-    @field_validator("eia860m")
+    @field_validator("eia860m_year_months")
     @classmethod
-    def check_eia860m_year_month(cls, eia860m: bool) -> bool:
-        """Check 860m date-year is exactly one year after most recent working 860 year.
+    def add_other_860m_years(cls, v, info: FieldValidationInfo) -> list[str]:
+        """Find extra years from EIA860m if applicable.
 
-        Args:
-            eia860m: True if 860m is requested.
+        There's a gap in reporting (after the new year but before the EIA 860 early
+        release data in June) when we rely on two years worth of EIA860m data instead
+        of just one. This function adds the last available month_year values for each
+        year of 860m data that is not yet available in 860.
 
-        Returns:
-            eia860m: True if 860m is requested.
-
-        Raises:
-            ValueError: the 860m date is within 860 working years.
         """
-        eia860m_year = pd.to_datetime(cls.eia860m_year_month).year
-        expected_year = max(cls.data_source.working_partitions["years"]) + 1
-        if eia860m and (eia860m_year != expected_year):
-            raise AssertionError(
-                f"Attempting to integrate an eia860m year "
-                f"({eia860m_year}) from {cls.eia860m_year_month} not immediately following"
-                f"the eia860 years: {cls.data_source.working_partitions['years']}. "
-                f"Consider switching eia860m parameter to False."
-            )
-        return eia860m
+        if info.data["eia860m"]:
+            all_eia860m_years = {
+                date.split("-")[0] for date in info.data["all_eia860m_year_months"]
+            }
+            # The years in 860m that are not in 860
+            extra_eia860m_years = {
+                year
+                for year in all_eia860m_years
+                if int(year) not in info.data["years"]
+            }
```

Comment:
Something I think this might permit, but which we probably wouldn't want is if someone put in an early EIA-860M year-month, and then only processed a single year of annual EIA-860 data from after that early EIA-860M date, I think it would try to add in EIA-860M dates from before the included EIA-860 data, as well as after it.

Reply:
In playing with the example test for overlapping years in the EIA-860M and EIA-860 settings, this bug was exercised -- it added **all** of the EIA-860M years before the test EIA-860 year, as well as after.